### PR TITLE
Slider modify update

### PR DESCRIPTION
**Pull request by davywentworth**

### CODE REVIEW

**Comment by theElectricMiracleCat** [comment]
can't seem to find the offset variable in the options hash after it's passed in through the behavior API. either it's unused or the framework is doing some magic that i'm unaware of. i'm curious to know what that magic is

otherwise, everything lgtm

**Comment by davywentworth** [comment]
see how Slider.Modify `Extends` Slider? `offset` lives on Slider. That Mootools classes at work.

**Comment by theElectricMiracleCat** [comment]
ah of course... missed that. thanks for explaining

File: js/Source/Slider.Modify/Behavior.Slider.Modify.js [hunk 1]
```diff
@@ -31,7 +32,11 @@ Behavior.addGlobalFilter('Slider.Modify', {
     // slideFill is optional
     var slideFill = api.get('fill') ? api.getElement('fill') : null;
     var knob = api.getElement('knob');
+    var offset = api.get('offset');
```
Comment (anutron) [inline js/Source/Slider.Modify/Behavior.Slider.Modify.js]: `.getAs(Number, 'offset')`

File: js/Source/Slider.Modify/Behavior.Slider.Modify.js [hunk 2]
```diff
@@ -31,7 +32,11 @@ Behavior.addGlobalFilter('Slider.Modify', {
...
     var targets = api.getAs(Array, 'targets');
+    var moveClassTargets;
+    if(api.get('moveClassTargets')) moveClassTargets = api.getElements('moveClassTargets');
+    var moveClass = api.get('moveClass');
```
Comment (anutron) [inline js/Source/Slider.Modify/Behavior.Slider.Modify.js]: I wouldn't declare this variable if you're only using it once. Just put the api call down on line 55...

Up to you. Not a big deal.

Comment (anutron) [inline js/Source/Slider.Modify/Behavior.Slider.Modify.js]: Ditto about offset above; you only use it once; just put it inline (example on line 50)

File: js/Source/Slider.Modify/Slider.Modify.js [hunk 3]
```diff
@@ -75,9 +76,22 @@ Slider.Modify = new Class({
     }, this);
   },
 
+  addMoveClass: function(){
+    if(this.options.moveClass && this.options.moveClassTargets){
+      this.options.moveClassTargets.addClass(this.options.moveClass)
+    }
+  },
```
Comment (anutron) [inline js/Source/Slider.Modify/Slider.Modify.js]: I don't think this really belongs here. Your class is nicely abstract. You already have an event model here. This belongs in the behavior in an `onMove` event handler.

File: js/Source/Slider.Modify/Slider.Modify.js [hunk 4]
```diff
@@ -75,9 +76,22 @@ Slider.Modify = new Class({
...
-    if (this.slideFill) this.slideFill.setStyle('width', percentage+"%");
+    var knobWidth = this.knob.getSize()[this.axis].toInt()/2;
+    var pos = 'top';
+    var dimension = 'height';
+    if (this.axis == 'x'){
+      pos = 'left';
+      dimension = 'width';
+    }
+    var knobCenter = this.knob.getStyle(pos).toInt() + knobWidth;
+    if (this.slideFill) this.slideFill.setStyle(dimension, knobCenter+"px");
```
Comment (anutron) [inline js/Source/Slider.Modify/Slider.Modify.js]: you don't need px. MooTools takes integers.

File: js/Source/Slider.Modify/Slider.Modify.js (hunk 3 above)
Comment (anutron) [inline js/Source/Slider.Modify/Slider.Modify.js]: It means, among other things, that you can remove a lot of code here around passing these options around, and it means that the somewhat esoteric use case (of adding classes) isn't conditionalized here (line 80). Just way cleaner IMHO.

**Comment by davywentworth** [comment]
@anutron i'll adjust after i write the documentation

File: js/Source/Slider.Modify/Slider.Modify.js (hunk 3 above)
Comment (davywentworth) [inline js/Source/Slider.Modify/Slider.Modify.js]: to me, it is the same as the code in the 'onComplete' handler. this whole class is very specific to this use case. what i should really do (later) is refactor to make a 'Slider.Fill' class, and then have this specialty class inherit from that.

Comment (anutron) [inline js/Source/Slider.Modify/Slider.Modify.js]: If you were to remove your focus from this particular use case you're working on how would you tackle this in a more generic fashion? For example, right now you are adding a class to an arbitrary group of elements specified in the options. Should you remove the class when movement stops? What if you need to do something other than adding a class? What if you want to transition a property? Create an element? Alter a style? And would you want to undo that (or stop doing it) when movement is complete?

Your implementation here is specific to one screen you're trying to tackle. That kind of use case either belongs in the instantiation of the class and its esoteric DOM (and, in Behavior land, that means in the Behavior), or it should be generic, which is basically what events are for.

I'm not saying you can't ship this. Just that you're basically pushing a function (add a class to a collection of elements) into a class that isn't about that. It's about being a slider.

**Pull request merged**